### PR TITLE
Refactor more variable names.

### PR DESCRIPTION
**Pull request by bdice**
## Description
This PR is a continuation of #355, with mostly changes that I requested and some that I didn't. I forgot to "submit" my PR review so these didn't make it into that PR. Lots of documentation fixes here.

## Motivation and Context
Consistency is the ~hobgoblin~ saving grace of copy-pasting minds.

## How Has This Been Tested?
Existing tests have been updated to use more consistent names.

## Types of changes

- [x] Mostly docs and minor refactoring 📝

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/glotzerlab/freud/blob/master/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] I have updated the documentation (if relevant).
- [x] I have added tests that cover my changes (if relevant).
- [x] All new and existing tests passed.
- [ ] I have updated the [Changelog](https://github.com/glotzerlab/freud/blob/master/ChangeLog.md).

### CODE REVIEW

**Comment by bdice** [comment]
@vyasr This will be terrible to review -- I was as meticulous as I could be while making the changes, and we have agreed on virtually all of these name changes previously, in #355 or in person. Feel free to just take a quick glance if you don't want to get stuck in the weeds here.

File: cpp/cluster/Cluster.h [hunk 1]
```diff
@@ -83,7 +83,7 @@ class Cluster
     }
 
 private:
-    float m_rcut;                 //!< Maximum r at which points will be counted in the same cluster
+    float m_r_cut;                 //!< Maximum r at which points will be counted in the same cluster
```
Comment (vyasr) [inline cpp/cluster/Cluster.h]: Change to r_max.

File: cpp/density/GaussianDensity.h [hunk 2]
```diff
@@ -61,7 +61,7 @@ class GaussianDensity
 private:
     box::Box m_box;                               //!< Simulation box where the particles belong
     vec3<unsigned int> m_width;                   //!< Num of bins on each side of the cube
-    float m_rcut;                                 //!< Max r at which to compute density
+    float m_r_cut;                                 //!< Max r at which to compute density
```
Comment (vyasr) [inline cpp/density/GaussianDensity.h]: Change to `m_r_max`.

File: cpp/density/LocalDensity.h [hunk 3]
```diff
@@ -52,7 +52,7 @@ class LocalDensity
 
 private:
     box::Box m_box;       //!< Simulation box where the particles belong
-    float m_rcut;         //!< Maximum neighbor distance
+    float m_r_cut;         //!< Maximum neighbor distance
```
Comment (vyasr) [inline cpp/density/LocalDensity.h]: Change to `r_max`.

File: cpp/locality/AABBQuery.cc [hunk 4]
```diff
@@ -73,7 +73,7 @@ void AABBIterator::updateImageVectors(float r_max, bool _check_r_max)
             || (periodic.y && nearest_plane_distance.y <= r_max * 2.0)
             || (!box.is2D() && periodic.z && nearest_plane_distance.z <= r_max * 2.0))
         {
-            throw std::runtime_error("The AABBQuery rcut is too large for this box.");
+            throw std::runtime_error("The AABBQuery r_cut is too large for this box.");
```
Comment (vyasr) [inline cpp/locality/AABBQuery.cc]: `r_max`

File: cpp/locality/NearestNeighbors.cc [hunk 5]
```diff
@@ -59,7 +59,7 @@ void NearestNeighbors::compute(const box::Box& box, const vec3<float>* ref_pos,
     ThreadBondVector bond_vectors;
 
     m_lc->computeCellList(m_box, ref_pos, num_ref);
-    const float rcutsq(m_lc->getCellWidth() * m_lc->getCellWidth());
+    const float r_cut_sq(m_lc->getCellWidth() * m_lc->getCellWidth());
```
Comment (vyasr) [inline cpp/locality/NearestNeighbors.cc]: `r_max_sq`

File: cpp/order/SolLiq.h [hunk 6]
```diff
@@ -57,15 +57,15 @@ class SolLiq
     }
 
     //! Reset the simulation box size
-    void setClusteringRadius(float rcut_cluster)
+    void setClusteringRadius(float r_cut_cluster)
```
Comment (vyasr) [inline cpp/order/SolLiq.h]: `r_max_cluster`

File: cpp/order/SolLiq.h [hunk 7]
```diff
@@ -57,15 +57,15 @@ class SolLiq
...
     {
-        if (rcut_cluster < m_rmax)
+        if (r_cut_cluster < m_r_max)
             throw std::invalid_argument(
-                "SolLiq requires that rcut_cluster must be greater than rcut (for local env).");
-        // May not be necessary if std::max(m_rmax, m_rmax_cluster) is used to rebuild cell list here, and in
+                "SolLiq requires that r_cut_cluster must be greater than r_cut (for local env).");
```
Comment (vyasr) [inline cpp/order/SolLiq.h]: `r_max_cluster`, `r_max`

**Review by vyasr — CHANGES_REQUESTED** [review]
Pretty much just need to do some more `r_cut`->`r_max` renames then it's good.

**Pull request merged**